internal/k8s/controller.go

```diff
@@ -1454,25 +1469,35 @@ func (lbc *LoadBalancerController) updateRegularIngressStatusAndEvents(ingConfig
 	}
 }
 
-func (lbc *LoadBalancerController) updateTransportServerEvents(tsConfig *TransportServerConfiguration, operationErr error) {
+func (lbc *LoadBalancerController) updateTransportServerStatusAndEvents(tsConfig *TransportServerConfiguration, operationErr error) {
 	eventTitle := "AddedOrUpdated"
 	eventType := api_v1.EventTypeNormal
 	eventWarningMessage := ""
+	state := conf_v1.StateValid
 
 	if len(tsConfig.Warnings) > 0 {
 		eventType = api_v1.EventTypeWarning
 		eventTitle = "AddedOrUpdatedWithWarning"
 		eventWarningMessage = fmt.Sprintf("with warning(s): %s", formatWarningMessages(tsConfig.Warnings))
+		state = conf_v1.StateInvalid
```

Comment:
@LorcanMcVeigh ^ ?

Reply:
I think it's ok to leave the state as either `Valid` or `Invalid` because having the state be `Warning` doesn't tell whether the resource is valid or not. But leaving the state as `Valid` or `Invalid` and having the reason next to it with the message `AddedOrUpdatedWithWarning` would make things a lot clearer 
```$ kubectl get transportserver
  NAME      STATE   REASON                      IP    PORTS   AGE
  dns-tcp   Valid   AddedOrUpdatedWithWarning                 47m
  ```
  This is a lot clearer, in relation to the state of the TransportServer than 
  ```$ kubectl get transportserver
  NAME      STATE     REASON                      IP    PORTS   AGE
  dns-tcp   Warning   AddedOrUpdatedWithWarning                 47m
  ```

Reply:
I think this is something we can change for VirtualServers and VirtualServerRoutes as well, but at another time

Reply:
I've compromised and gone with StateWarning. I'm still not sure it's correct though.